### ISSUE TITLE
[Java] Component types on generic arrays are lost during (de-)serialization

### ISSUE THREAD

**Issue by mtf90** [apache/incubator-fury#1323]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-fury/issues) and found no similar issues.

### Version

* 0.5.0-SNAPSHOT from the nightly builds (current HEAD is a80dce6)
* OS: Linux x64
* Java: fails on JDK 11/17/21 

### Component(s)

Java

### Minimal reproduce step

```java
public class App {

    public static void main(String[] args) {
        final Fury fury = Fury.builder().requireClassRegistration(false).build();
        final ArrayWrapper<String> wrapper = new ArrayWrapper<>(String.class, 2);

        wrapper.array[0] = "Hello";

        final byte[] bytes = fury.serialize(wrapper);
        final ArrayWrapper<String> deserialized = (ArrayWrapper<String>) fury.deserialize(bytes);

        deserialized.array[1] = "World";
    }

    static class ArrayWrapper<T> {

        private final T[] array;

        public ArrayWrapper(Class<T> clazz, int capacity) {
            this.array = (T[]) Array.newInstance(clazz, capacity);
        }
    }
}
```

### What did you expect to see?

A successfull run of the program

### What did you see instead?

```
Exception in thread "main" java.lang.ClassCastException: class [Ljava.lang.Object; cannot be cast to class [Ljava.lang.String; ([Ljava.lang.Object; and [Ljava.lang.String; are in module java.base of loader 'bootstrap')
```

### Anything Else?

It appears that the information about the component type (here `String.class`) are lost and the (de-) serializer simply assumes "Generic Type? => Object!". The problem here is that one cannot simply replace the `Array.newInstance(clazz, capacity);` with `new Object[capacity];` because then the `wrapper.array[0] = "Hello";` statement fails with the same error.

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by chaokunyang** [comment]
Thanks @mtf90 , I can reproduce this bug locally. This is a bug in fury. Fury take `Object[]` as a final type, it skip writing class info for such types, so the deserialization got wrong type. I fixed it in #1324. Thanks again for reporting this bug.

**Comment by mtf90** [comment]
Seems like fury is not only blazing fast with serialization but also with bug fixes :laughing:. Thank you very much for the quick response. Looking forward to test the nightly once this is merged.
